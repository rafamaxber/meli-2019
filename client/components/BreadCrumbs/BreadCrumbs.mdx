---
name: BreadCrumbs
menu: Components
---

import { Playground, Props } from 'docz'
import { MainThemeProvider } from '../Layouts/ThemeProvider.tsx'

import BreadCrumbs from './BreadCrumbs.tsx'

# BreadCrumbs

## Properties

<Props of={BreadCrumbs} />

## Basic usage

<Playground>
  <MainThemeProvider>
    <BreadCrumbs steps={['step 1', 'step 2', 'step 3']} />
  </MainThemeProvider>
</Playground>

