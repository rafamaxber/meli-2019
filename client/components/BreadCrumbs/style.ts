import styled from 'styled-components';

export const BreadCrumb = styled.nav`
  font-size: 14px;
  padding: ${({ theme }) => theme.paddingBase} 0;
  color: ${({ theme }) => theme.lv3Color};
`;

export const BreadCrumbList = styled.ul`
  margin: 0 0 0 0;
  padding: 0 0 0 0;
`;

export const BreadCrumbItem = styled.li`
  display: inline-block;
  list-style-type: none;
  vertical-align: middle;
  cursor: default;

  :last-child {
    font-weight: bold;
  }

  svg {
    margin: 0 5px;
    vertical-align: bottom;
    display: inline-block;
  }
`;
