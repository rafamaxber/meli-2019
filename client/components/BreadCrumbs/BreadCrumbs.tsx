import * as Styled from './style'
import * as Common from '../Common/main.styled'

interface BreadCrumbs {
  /**
   * Set this to show breadcrumb items
   */
  steps: string[]
}

const Arrow = ({ show }) => show ? (
  <svg width="15" height="15" viewBox="0 0 18 18" xmlns="http://www.w3.org/2000/svg">
    <g fill="#999999" fillRule="evenodd">
      <path d="M6.646 5.354l4 4 .354.353.707-.707-.353-.354-4-4L7 4.293 6.293 5z"></path>
      <path d="M7.354 13.354l4-4L11.707 9 11 8.293l-.354.353-4 4-.353.354.707.707z"></path>
    </g>
  </svg>
) : null;

function BreadCrumbs({ steps }: BreadCrumbs): JSX.Element {
  if (!steps || !steps.length) return null
  const stepsList = steps.map((item, index, array) => {
    return (
      <Styled.BreadCrumbItem key={index}>
        { item }
        <Arrow show={Boolean(array[index + 1])} />
      </Styled.BreadCrumbItem>)
  });

  return (
    <Common.Limit>
      <Styled.BreadCrumb aria-label="Breadcrumb">
        <Styled.BreadCrumbList>
          {stepsList}
        </Styled.BreadCrumbList>
      </Styled.BreadCrumb>
    </Common.Limit>
  )
}

export default BreadCrumbs
