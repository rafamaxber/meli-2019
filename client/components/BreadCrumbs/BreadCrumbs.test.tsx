import React from 'react';
import { render } from '@testing-library/react';
import BreadCrumbs from './BreadCrumbs';

describe('BreadCrumbs', () => {
  describe('smoke tests', () => {
    test('should render nothing', () => {
      const component = render(<BreadCrumbs />);

      expect(component).toMatchSnapshot();
    })

    test('should render component with breadcrumbs', () => {
      const component = render(<BreadCrumbs steps={['Step 1', 'Step 2', 'Step 3']}/>);

      expect(component).toMatchSnapshot();
    })
  });
})


