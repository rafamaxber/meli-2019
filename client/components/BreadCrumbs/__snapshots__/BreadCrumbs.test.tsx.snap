// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadCrumbs smoke tests should render component with breadcrumbs 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    .c0 {
  font-size: 14px;
  padding: 0;
}

.c1 {
  margin: 0 0 0 0;
  padding: 0 0 0 0;
}

.c2 {
  display: inline-block;
  list-style-type: none;
  vertical-align: middle;
  cursor: default;
}

.c2:last-child {
  font-weight: bold;
}

.c2 svg {
  margin: 0 5px;
  vertical-align: bottom;
  display: inline-block;
}

<div>
      <div
        class=""
      >
        <nav
          aria-label="Breadcrumb"
          class="c0"
        >
          <ul
            class="c1"
          >
            <li
              class="c2"
            >
              Step 1
              <svg
                height="15"
                viewBox="0 0 18 18"
                width="15"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  fill="#999999"
                  fill-rule="evenodd"
                >
                  <path
                    d="M6.646 5.354l4 4 .354.353.707-.707-.353-.354-4-4L7 4.293 6.293 5z"
                  />
                  <path
                    d="M7.354 13.354l4-4L11.707 9 11 8.293l-.354.353-4 4-.353.354.707.707z"
                  />
                </g>
              </svg>
            </li>
            <li
              class="c2"
            >
              Step 2
              <svg
                height="15"
                viewBox="0 0 18 18"
                width="15"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  fill="#999999"
                  fill-rule="evenodd"
                >
                  <path
                    d="M6.646 5.354l4 4 .354.353.707-.707-.353-.354-4-4L7 4.293 6.293 5z"
                  />
                  <path
                    d="M7.354 13.354l4-4L11.707 9 11 8.293l-.354.353-4 4-.353.354.707.707z"
                  />
                </g>
              </svg>
            </li>
            <li
              class="c2"
            >
              Step 3
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mainstyled__Limit-kgem6r-2 poVdl"
    >
      <nav
        aria-label="Breadcrumb"
        class="style__BreadCrumb-sc-1bo4fpd-0 eiYiIH"
      >
        <ul
          class="style__BreadCrumbList-sc-1bo4fpd-1 jvnaLM"
        >
          <li
            class="style__BreadCrumbItem-sc-1bo4fpd-2 bUXwIg"
          >
            Step 1
            <svg
              height="15"
              viewBox="0 0 18 18"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                fill="#999999"
                fill-rule="evenodd"
              >
                <path
                  d="M6.646 5.354l4 4 .354.353.707-.707-.353-.354-4-4L7 4.293 6.293 5z"
                />
                <path
                  d="M7.354 13.354l4-4L11.707 9 11 8.293l-.354.353-4 4-.353.354.707.707z"
                />
              </g>
            </svg>
          </li>
          <li
            class="style__BreadCrumbItem-sc-1bo4fpd-2 bUXwIg"
          >
            Step 2
            <svg
              height="15"
              viewBox="0 0 18 18"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                fill="#999999"
                fill-rule="evenodd"
              >
                <path
                  d="M6.646 5.354l4 4 .354.353.707-.707-.353-.354-4-4L7 4.293 6.293 5z"
                />
                <path
                  d="M7.354 13.354l4-4L11.707 9 11 8.293l-.354.353-4 4-.353.354.707.707z"
                />
              </g>
            </svg>
          </li>
          <li
            class="style__BreadCrumbItem-sc-1bo4fpd-2 bUXwIg"
          >
            Step 3
          </li>
        </ul>
      </nav>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`BreadCrumbs smoke tests should render nothing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
